pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -617,6 +651,50 @@ func (r *ReconcileLogStorage) deleteInvalidECKManagedPublicCertSecret(ctx contex
 	return r.client.Delete(ctx, secret)
 }
 
+func (r *ReconcileLogStorage) getElasticsearchCertificateSecrets(ctx context.Context) (*corev1.Secret, *corev1.Secret, *corev1.Secret, error) {
+	svcDNSNames := dns.GetServiceDNSNames(render.ElasticsearchServiceName, render.ElasticsearchNamespace, r.clusterDomain)
+
+	// Get the secret - might be nil
+	certSecret, err := utils.GetSecret(ctx, r.client, render.TigeraElasticsearchCertSecret, rmeta.OperatorNamespace())
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// Ensure that cert is valid.
+	certSecret, err = utils.EnsureCertificateSecret(render.TigeraElasticsearchCertSecret, certSecret, "tls.key", "tls.crt", rmeta.DefaultCertificateDuration, svcDNSNames...)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	certSecretESCopy := rsecret.CopyToNamespace(render.ElasticsearchNamespace, certSecret)[0]
+
+	// Get the pub secret - might be nil
+	pubSecret, err := utils.GetSecret(ctx, r.client, relasticsearch.PublicCertSecret, render.ElasticsearchNamespace)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	if pubSecret != nil {
+		operatorManaged, err := utils.IsOperatorManaged(certSecret, "tls.crt")
```

Comment:
Should this be using `pubSecret` instead of `certSecret`

Reply:
It's actually supposed to be certSecret. certSecret is what the user would create and the ECK operator creates pubSecret. What this is saying is that if this is operator managed and the dns name is wrong, delete the public secret so it can get recreated.

Reply:
Could you include a comment on the if block that is the intention here?

Also how does this work then when a user provides the certSecret (after first using the auto-generated one)? Is it possible that we need to delete the pubSecret so it gets recreated?

Reply:
I think you've pointed out an existing bug in the code. This is the logic that was there originally so I'm going to leave it as is, but I've created this ticket https://tigera.atlassian.net/browse/SAAS-1365 to address this issue.

Reply:
I would still like a comment on why in this pubSecret if block that we're checking if the certSecret "IsOperatorManaged".

Reply:
I added the comment to the wrong section, just moved it.